src/prototype_creep_resources.js

```diff
@@ -486,7 +486,7 @@ Creep.prototype.transferMy = function() {
 };
 
 Creep.prototype.getEnergy = function() {
-  if (this.carry.energy == this.carryCapacity) {
+  if (this.carry.energy > 0) {
```

Comment:
Agreed, I think it's better for it to be a state system instead, where the creep gets energy until capacity, then spends energy until 0, and repeat. WDYT?